queens/iterators/control_variates_iterator.py

```diff
@@ -0,0 +1,246 @@
+    """
+
+    @log_init_args
+    def __init__(
+        self,
+        model,
+        control_variate,
+        parameters,
+        global_settings,
+        seed,
+        num_samples,
+        expectation_cv=None,
+        num_samples_cv=None,
+        use_optimal_num_samples=False,
+        cost_model=None,
+        cost_cv=None,
+    ):
+        """Initialize the control variates iterator.
+
+        Args:
+            model (Model):  The main model. Uncertainties are quantified for this model.
+            control_variate (Model):    The control variate - a model that approximates the main
+            model and is cheaper to evaluate.
+            parameters (Parameters):    Parameters to use for evaluation.
+            global_settings (GlobalSettings):   Global settings to use for evaluation.
+            seed (int): Seed for random samples.
+            num_samples (int):  Number of samples to evaluate for estimation.
+            expectation_cv (float): Here you can pass expected values of the control variate.
+            num_samples_cv (int):   Number of MC samples to use for computing the expectation of
+                                    the control variate. Only used if the expectation of the control
+                                    variate is not passed.
+            cost_model (float): Cost of evaluating the model.
+            cost_cv (float):    Cost of evaluating the control variate.
+            use_optimal_num_samples (bool):  Determines if the iterator calculates and uses
+                                                the optimal number of samples.
+
+        Raises:
+            ValueError: If model is None.
+                        If control_variable is None.
+                        If neither the expectation nor the number of samples of the control
+                        variate are given and the optimal number of samples is not used.
+                        If the optimal number of samples are used and the costs of the models are
+                        not provided.
+        """
+        if model is None or control_variate is None:
+            raise ValueError("A model and a control variate have to be given!")
+
+        if expectation_cv is None and num_samples_cv is None and not use_optimal_num_samples:
+            raise ValueError(
+                """expectation_cv or num_samples_cv has to be given, when not using
+                             the optimal number of samples.
+                             """
+            )
+
+        if use_optimal_num_samples and (cost_model is None or cost_cv is None):
+            raise ValueError(
+                """model costs have to be given if you want to use the optimal number
+                    of samples
+                """
+            )
+
+        # Initialize parent iterator with no model.
+        super().__init__(None, parameters, global_settings)
+        self.model = model
+        self.control_variate = control_variate
```

Comment:
```suggestion
        super().__init__(model, parameters, global_settings)
        self.control_variate = control_variate
```
Now that we have split the models list into model and control variate, we can actually pass the main model to the Iterator parent class.